internal/distlock/.github/workflows/main.yml

```diff
@@ -0,0 +1,21 @@
+---
```

Comment:
I'm not sure if you want to keep these. 

I could not get docker-compose to work for the life of me in the GH-Actions environments. 

If you want to keep this just for completeness that's fine too, I don't think gh actions searches recursively for a .github folder and these files won't mess up the current actions, right?